app/utils/MessageRenderer.scala

```diff
@@ -21,8 +21,15 @@ import play.api.i18n.Messages
 
 object MessageRenderer {
   def getMessage(msgKey: String, msgArgs: Any*)(implicit messages: Messages, user: UserRequest[_]): String = {
+
     if (user.isAgent){
-      messages.apply(s"agent.$msgKey", msgArgs: _*)
+      if(user.session.get(SessionKeys.whoPlannedToSubmitVATReturn).contains("client") ||
+        (user.session.get(SessionKeys.whoPlannedToSubmitVATReturn).contains("agent") && user.session.get(SessionKeys.causeOfLateSubmissionAgent).contains("client"))){
```

Comment:
So can we not store the 'user.session.get(SessionKeys.whoPlannedToSubmitVATReturn)' in a String val and do the 'contains' as 'client' or 'agent'.

Reply:
Made that change locally yesterday, makes it easier to read 🙂